nuxeo-page-provider-display-behavior.html

```diff
@@ -514,15 +514,37 @@
       _fetchRange(firstIndex, lastIndex, clear) {
         if (this._hasPageProvider()) {
 
+          if (firstIndex === 0) {
+            lastIndex = this._pageSize - 1;
+          }
+
           if (!firstIndex && !lastIndex && !clear) {
             firstIndex = 0;
-            lastIndex = this._pageSize;
+            lastIndex = this._pageSize - 1;
             clear = true;
```

Comment:
Perhaps we could replace these two if blocks by something shorter like:
```
firstIndex = firstIndex || 0;
lastIndex = (firstIndex === 0 || lastIndex === undefined) ? this._pageSize - 1 : lastIndex;
clear = (clear === undefined) ? true : clear;
```

Reply:
yes, but harder to debug :P

Reply:
plus there's something wrong in your refactoring, scrolling is completely broken :D

Reply:
I didn't actually tested the code, it was just a simplification assuming the previous if was checking for `undefined`, which it seems it isn't... 😬